test/common/thread_local/thread_local_impl_test.cc

```diff
@@ -82,6 +82,40 @@ TEST_F(ThreadLocalInstanceImplTest, All) {
   tls_.shutdownThread();
 }
 
+// Validate ThreadLocal::runOnAllThreads behavior with all_thread_complete call back.
+TEST_F(ThreadLocalInstanceImplTest, RunOnAllThreads) {
+
+  SlotPtr tlsptr = tls_.allocateSlot();
+
+  EXPECT_CALL(thread_dispatcher_, post(_));
+  EXPECT_CALL(main_dispatcher_, post(_));
+
+  // Ensure that the thread local call back and all_thread_complete call back are called.
+  struct thread_local_verification_info {
+    std::atomic<uint64_t> thread_local_calls_{0};
+    std::condition_variable condvar_;
+    std::mutex condvar_mutex_;
+    bool all_threads_complete_;
+  };
+
+  thread_local_verification_info thread_local_data;
```

Comment:
If you are going to define a type it should be called ThreadLocalVerificationInfo, but you don't need to define a type.  Just do:
```
struct {
  std::atomic<uint64_t> thread_local_calls_{0};
  std::condition_variable condvar_;
  std::mutex condvar_mutex_;
  bool all_threads_complete_;
} thread_local_data;
```